### PR TITLE
#430 Deprecate api-cc-infix add replacements for api-infix arrayAssertions

### PR DESCRIPTION
**Pull request by Miftahunajat**
This PR add Deprecate api-cc-infix and add Replacewith for the expect for these files:
- [x] arrayAssertions

______________________________________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/ArrayAsIterableAssertionsSpec.kt [hunk 1]
```diff
@@ -28,25 +31,25 @@ class ArrayAsIterableAssertionsSpec : ch.tutteli.atrium.spec.integration.ArrayAs
...
-        fun arrayInt(plant: Assert<Array<Int>>) = plant.asIterable()
-        fun byteArray(plant: Assert<ByteArray>) = plant.asIterable()
-        fun charArray(plant: Assert<CharArray>) = plant.asIterable()
-        fun shortArray(plant: Assert<ShortArray>) = plant.asIterable()
-        fun intArray(plant: Assert<IntArray>) = plant.asIterable()
-        fun longArray(plant: Assert<LongArray>) = plant.asIterable()
-        fun floatArray(plant: Assert<FloatArray>) = plant.asIterable()
-        fun doubleArray(plant: Assert<DoubleArray>) = plant.asIterable()
-        fun booleanArray(plant: Assert<BooleanArray>) = plant.asIterable()
+        fun arrayInt(plant: Assert<Array<Int>>) = plant.asExpect<Array<out Int>, Assert<Array<out Int>>>().asList(ch.tutteli.atrium.api.infix.en_GB.o).asAssert()
```
Comment (robstoll) [inline apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/ArrayAsIterableAssertionsSpec.kt:34]: oh man, that looks ugly :laughing: especially because of `ch.tutteli.atrium.api.infix.en_GB.o`
however, if I checkout your branch and do the replacement, then I get
```
plant.asExpect<Array<out Int>, Assert<Array<out Int>>>().asList(o).asAssert()
```
which is much better. Can you please check if the replacement works for you; i.e. revert the changes in this file and do the replacements again.

Comment (Miftahunajat) [reply apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/ArrayAsIterableAssertionsSpec.kt:34]: Yeah I know it's a bit ugly.
Because I can't automatically replace it with `plant.asExpect<Array<out Int>, Assert<Array<out Int>>>().asList(o).asAssert()` I will check this out.

Comment (robstoll) [reply apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/ArrayAsIterableAssertionsSpec.kt:34]: Don't spend too much time, it is fine like this as well. I was merely curious why it did not work for you since it worked for me.

File: apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/charSequenceAssertions.kt [hunk 2]
```diff
@@ -49,6 +49,15 @@ infix fun <T : CharSequence> Assert<T>.notTo(@Suppress("UNUSED_PARAMETER") conta
  * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
  * @throws IllegalArgumentException in case [expected] is not a [CharSequence], [Number] or [Char].
  */
+@Deprecated(
+    "Switch from Assert to Expect; will be removed with 1.0.0 -- see https://github.com/robstoll/atrium/releases/tag/v0.9.0#migration for migration hints and scripts.",
+    ReplaceWith(
```
Comment (robstoll) [inline apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/charSequenceAssertions.kt:54]: Please deprecate `to contain` and `notTo contain` above as well (almost everything needs to be deprecated, please ask if you are unsure about one). The equivalent call in the new infix API looks like the following `contains(o)`, `containsNot(o)`

File: apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/CharSequenceContainsContainsNotAssertionsSpec.kt [hunk 3]
```diff
@@ -29,9 +32,9 @@ class CharSequenceContainsContainsNotAssertionsSpec : ch.tutteli.atrium.spec.int
 
         private fun containsNot(plant: Assert<CharSequence>, a: Any, aX: Array<out Any>): Assert<CharSequence> {
             return if (aX.isEmpty()) {
-                plant containsNot a
+                plant.asExpect().containsNot(a).asAssert()
```
Comment (robstoll) [inline apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/CharSequenceContainsContainsNotAssertionsSpec.kt:35]: you forgot some deprecated things in this file, please check again

File: apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/charSequenceAssertions.kt [hunk 4]
```diff
@@ -180,6 +261,15 @@ infix fun <T : CharSequence> Assert<T>.endsWith(expected: CharSequence)
...
+        "this.asExpect().endsNotWith(expected).asAssert()",
+        "ch.tutteli.atrium.domain.builders.migration.asExpect",
+        "ch.tutteli.atrium.domain.builders.migration.asAssert",
+        "ch.tutteli.atrium.api.infix.en_GB.endsNotWith"
+    )
+)
```
Comment (robstoll) [inline apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/charSequenceAssertions.kt:272]: There are more functions further below which should be deprecated as well. `toBe Empty` and `notToBe Empty/Blank`. Note that the keywords `Empty` and `Blank` are lower case in the new infix API

File: apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/charSequenceContainsCheckers.kt [hunk 5]
```diff
@@ -17,6 +17,13 @@ import ch.tutteli.atrium.domain.creating.charsequence.contains.CharSequenceConta
  * @throws IllegalArgumentException In case [times] is smaller than zero.
  * @throws IllegalArgumentException In case [times] equals to zero; use [enthaeltNicht] instead.
  */
+@Deprecated(
+    "Switch from api-cc-en_GB to api-infix-en_GB; will be removed with 1.0.0 -- see https://github.com/robstoll/atrium/releases/tag/v0.9.0#migration for migration hints and scripts.",
```
Comment (robstoll) [inline apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/charSequenceContainsCheckers.kt:21]: please adopt the text (I would use a search&replace in the cc-infix folder):
It should be:
```suggestion
    "Switch from api-cc-infix-en_GB to api-infix-en_GB; will be removed with 1.0.0 -- see https://github.com/robstoll/atrium/releases/tag/v0.9.0#migration for migration hints and scripts.",
```

File: apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/charSequenceAssertions.kt (hunk 2 above)
Comment (Miftahunajat) [reply apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/charSequenceAssertions.kt:54]: The reason I'm unsure about deprecating the `to contain` and `notTo contain` is because there isn't such implementation in the `cc-en_GB` files. and I don't know that the API call equivalent the `contains(o)` and `containsNot(o)`. thank you for giving me such info.

**Comment by Miftahunajat** [comment]
> Do you plan to include all CharSequence related files in this PR, i.e. charSequenceContainsCreators and charSequenceContainsSearchBehaviours in addition? Would make sense but you can also do it in a next PR up to you (or remove all of them from this PR so that only array related files are in this PR)
> 
> Note also that you missed some of the CharSequence...Specs.

I will move it to another PR then.

File: apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/charSequenceAssertions.kt (hunk 2 above)
Comment (robstoll) [reply apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/charSequenceAssertions.kt]: Just ask in case you are uncertain, I am here to answer :slightly_smiling_face: (it would be too much work to write down every detail which could cause problems beforehand).
As hint, usually the structure of the files in the api-infix is the same as in api-cc-infix. This way you could have seen that `to contain` in apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/charSequenceAssertions.kt is now superseded by `contains(o)` in apis/infix-en_GB/atrium-api-infix-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/infix/en_GB/charSequenceAssertions.kt
I hope this hint helps for further deprecations, there are other changes in api-infix (e.g. list.get is different)
